onadata/apps/api/viewsets/open_data_viewset.py

```diff
@@ -84,10 +147,12 @@ def append_to_tableau_colulmn_headers(header, question_type=None):
                 'dataType': quest_type,
                 'alias': header
             })
+        # Remove metadata fields from the column headers
+        xform_headers = set(remove_metadata_fields(self.xform_headers))
```

Comment:
when does `xform_headers` return duplicates?

Reply:
I noticed that there could be duplicates for group data(not repeat data)
if the `specify other` question type is used.

This can be observed when you run the `test_column_headers_endpoint` and place a debugger here.
The schema generated without removing duplicate columns is
```code
[
  'transport_available_transportation_types_to_referral_facility_ambulance',
  'transport_available_transportation_types_to_referral_facility_bicycle',
  'transport_available_transportation_types_to_referral_facility_boat_canoe',
  'transport_available_transportation_types_to_referral_facility_bus',
  'transport_available_transportation_types_to_referral_facility_donkey_mule_cart',
  'transport_available_transportation_types_to_referral_facility_keke_pepe',
  'transport_available_transportation_types_to_referral_facility_lorry',
  'transport_available_transportation_types_to_referral_facility_motorbike',
  'transport_available_transportation_types_to_referral_facility_taxi',
  'transport_available_transportation_types_to_referral_facility_other',
  'transport_available_transportation_types_to_referral_facility_other',
  'transport_loop_over_transport_types_frequency_ambulance_frequency_to_referral_facility',
  'transport_loop_over_transport_types_frequency_bicycle_frequency_to_referral_facility',
  'transport_loop_over_transport_types_frequency_boat_canoe_frequency_to_referral_facility',
  'transport_loop_over_transport_types_frequency_bus_frequency_to_referral_facility',
  'transport_loop_over_transport_types_frequency_donkey_mule_cart_frequency_to_referral_facility',
  'transport_loop_over_transport_types_frequency_keke_pepe_frequency_to_referral_facility',
  'transport_loop_over_transport_types_frequency_lorry_frequency_to_referral_facility',
  'transport_loop_over_transport_types_frequency_motorbike_frequency_to_referral_facility',
  'transport_loop_over_transport_types_frequency_taxi_frequency_to_referral_facility',
  'image1',
  'meta_instanceID',
  '_id'
]
```

This introduces an extra column, making column headers more than row headers.
This difference is what I am avoiding

Reply:
make sense, can you add a note to the comment explaining this, something like, "calling `set` to remove duplicates in group data"